test/stripe/file_test.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require ::File.expand_path("../../test_helper", __FILE__)
+
+module Stripe
+  class FileTest < Test::Unit::TestCase
+    should "be listable" do
+      files = Stripe::File.list
+      assert_requested :get, "#{Stripe.api_base}/v1/files"
+      assert files.data.is_a?(Array)
+      assert files.data[0].is_a?(Stripe::File)
+    end
+
+    should "be retrievable" do
+      file = Stripe::File.retrieve("file_123")
+      assert_requested :get, "#{Stripe.api_base}/v1/files/file_123"
+      assert file.is_a?(Stripe::File)
+    end
+
+    context ".create" do
+      setup do
+        # We don't point to the same host for the API and uploads in
+        # production, but `stripe-mock` supports both APIs.
+        Stripe.uploads_base = Stripe.api_base
+
+        # Set `api_base` to `nil` to ensure that these requests are _not_ sent
+        # to the default API hostname.
+        Stripe.api_base = nil
```

Comment:
This is a bit awkward, but it's necessary because if `api_base` and `uploads_base` have the same value, then `assert_requested` is not actually asserting that the request is for the correct host.

Reply:
Good point. It may be worth adding a comment that notes that `api_base` will be reset between every test case so this won't mutate the global state of test runs that come after it.